Issue #512: ENS for addresses

jahabeebs opened the pull request:
closes #512 

## Description
- I replaced returnWalletAddress with a new useAddress hook everywhere I could. There were some places (like in AuctionBlock) where I couldn't replace it because we can't call a hook from there and I can't test the auction stuff (I'd have to refactor it to get the address passed in probably)

Some potential issues:
1) we'll probably get rate limited pretty quickly with our public read only RPC, should we switch it to a paid provider? The requirements said use a public one but if the call fails we'll fall back to just returning the address
2) the leaderboard has some latency rendering if we're going to check every address for ENS (see video)--should we just return the addresses for it instead?

## Screenshots

Testing with hardcoded account value

<img width="994" alt="test ens" src="https://github.com/open-dollar/od-app/assets/47253537/ba7407a1-48d6-4a40-b5cb-99e3a9dd543d">

Leaderboard (no ENS, either they don't have one or are getting rate limited by the rpc and so it's just returning the user's address)

https://github.com/open-dollar/od-app/assets/47253537/181e6613-22ac-4a5c-9c47-9f092734abbd

pi0neerpat reviewed (changes requested):
- [ ] Move all RPC urls to `src/chains.ts` and export them as constants eg `RPC_URL_OPTIMISM`
- [ ] The Ethereum RPC for Bridge (`src/utils/tokens.ts`) and ENS features should now be the same 
- [ ] Patrick will provide the private Ethereum RPC
- [ ] Remove wallet icon. Move Blockie + Address/ENS to the wallet button 
![image](https://github.com/open-dollar/od-app/assets/35622595/c747a953-ca98-4563-83d0-93ec059b4d00)

jahabeebs commented:
@pi0neerpat 

- [X] Move all RPC urls to `src/chains.ts` and export them as constants eg `RPC_URL_OPTIMISM`
- [X] The Ethereum RPC for Bridge (`src/utils/tokens.ts`) and ENS features should now be the same 
- [X] Patrick will provide the private Ethereum RPC
- [ ] Remove wallet icon. Move Blockie + Address/ENS to the wallet button 
- Added some variables we were missing to .env.example
- Updated od-app and od-app-testnet in vercel to have the new REACT_APP_RPC_URL_ETHEREUM

Will do as part of https://github.com/open-dollar/od-app/issues/582#event-13206126152
